src/parallel/HighsTaskExecutor.h

```diff
@@ -30,9 +30,12 @@ class HighsTaskExecutor {
  public:
   using cache_aligned = highs::cache_aligned;
   struct ExecutorHandle {
-    cache_aligned::shared_ptr<HighsTaskExecutor> ptr{nullptr};
+    HighsTaskExecutor* ptr{nullptr};
+    bool isMain{false};
+    bool isStopped{false};
```

Comment:
I would move `isStopped` into `HighsTaskExecutor` and wrap it in an `std::atomic<bool>`.